crates/fj-host/src/host.rs

```diff
@@ -1,31 +1,119 @@
-use crossbeam_channel::Receiver;
+use std::thread;
 
-use crate::{Error, Evaluator, Model, ModelEvent, Watcher};
+use crossbeam_channel::{unbounded, Receiver, Sender};
+use fj_operations::shape_processor::ShapeProcessor;
+use winit::event_loop::EventLoopProxy;
 
-/// A Fornjot model host
+use crate::{HostCommand, HostHandle, Model, ModelEvent, Watcher};
+
+// Use a zero-sized error type to silence `#[warn(clippy::result_large_err)]`.
+// The only error from `EventLoopProxy::send_event` is `EventLoopClosed<T>`,
+// so we don't need the actual value. We just need to know there was an error.
+/// Error type for sending on a channel with a closed event loop
+pub struct EventLoopClosed;
+
+/// A Fornjot model host that runs in the background
 pub struct Host {
-    evaluator: Evaluator,
-    _watcher: Watcher,
+    shape_processor: ShapeProcessor,
+    event_loop_proxy: EventLoopProxy<ModelEvent>,
+    command_tx: Sender<HostCommand>,
+    command_rx: Receiver<HostCommand>,
 }
 
 impl Host {
-    /// Create a new instance of `Host`
-    ///
-    /// This is only useful, if you want to continuously watch the model for
-    /// changes. If you don't, just keep using `Model`.
-    pub fn from_model(model: Model) -> Result<Self, Error> {
-        let watch_path = model.watch_path();
-        let evaluator = Evaluator::from_model(model);
-        let watcher = Watcher::watch_model(watch_path, &evaluator)?;
-
-        Ok(Self {
-            evaluator,
-            _watcher: watcher,
-        })
+    /// Create a new `Host` that will process models for an event loop.
+    pub fn new(
+        shape_processor: ShapeProcessor,
+        event_loop_proxy: EventLoopProxy<ModelEvent>,
+    ) -> Self {
+        let (command_tx, command_rx) = unbounded();
```

Comment:
Tiny nitpick: `unbounded()` doesn't really tell you what's happening here, unless you're already familiar with how channels work. This would be a bit more clear, I think:
```suggestion
        let (command_tx, command_rx) = crossbeam_channel::unbounded();
```